students/Net_Michael/lesson3/mailroom.py

```diff
@@ -0,0 +1,147 @@
+#!/usr/bin/env python
+# coding: utf-8
+
+# In[1]:
+
+
+# Part I
+
+import sys  
+
+def sort_key(donor_db2):
+        return donor_db2[0].split(" ")[1]
+
+# update the list of donors and donation
+def add_donor():
+    new_donor = input("Name of donor?").title()
+    if new_donor not in donor_name:
+        donor_name.append(new_donor)  
+        new_donation1 = input("Amount of donation by {}?".format(new_donor))
+        idx = len(donor_name)-1
+        new_donation1 = float(new_donation1)
+        donation[idx].append(new_donation1)
+    elif new_donor in donor_name:
+        idx = donor_name.index(new_donor)
+        new_donation1 = float(input("Amount of donation?".format(new_donor)))
+        donation[idx].append(new_donation1)
+    print(f"{new_donor} has made a new donation of {new_donation1}")
+    return(new_donor, new_donation1)
+
+
+# compile donors and donation information
+def donors_info():
+    
+    mm = len(donor_name)
+    donor_db2 = [''] * mm
+    
+    for j in range(mm):
+        donor_db2[j] = (donor_name[j], donation[j])
+        
+    donor_db2 = sorted(donor_db2, key=sort_key)
+    return(donor_db2)
```

Comment:
what is the purpose of this function? highly recommend using common python patterns for better code readability